string-manipulation-3/src/is-palindromic.js

```diff
@@ -0,0 +1,41 @@
+/* eslint-disable no-unused-vars */
+function isPalindromic(string) {
+  var original;
+  for (var i = 0; i < string.length; i++) {
+    if (string[i] !== ' ') {
+      original += string[i];
+    }
+  }
+  var palindrome;
+  for (var x = string.length - 1; x >= 0; x--) {
+    if (string[x] !== ' ') {
+      palindrome += string[x];
+    }
+  }
+  if (original === palindrome) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+/* can't understand why this won't work */
+function isPalindromicAlt(string) {
+  var original = [];
+  for (var y = 0; y < string.length; y++) {
+    if (string[y] !== ' ') {
+      original.push(string[y]);
+    }
+  }
+  var palindrome = [];
+  for (var z = string.length - 1; z >= 0; z--) {
+    if (string[z] !== ' ') {
+      palindrome.push(string[z]);
+    }
+  }
+  if (original === palindrome) {
```

Comment:
Because you are comparing arrays
Reference data types compare memory addresses, they are different memory addresses.

This would have worked

```
original.join('') === palindrome.join('')

Reply:
Cool, gotcha